tests/test_downloader_handlers.py

```diff
@@ -66,8 +72,11 @@ def test_disabled_handler(self):
         handlers = {'scheme': None}
         crawler = get_crawler(settings_dict={'DOWNLOAD_HANDLERS': handlers})
         dh = DownloadHandlers(crawler)
+        self.assertNotIn('scheme', dh._schemes)
+        for scheme in handlers: # force load handlers
+            dh._get_handler(scheme)
         self.assertNotIn('scheme', dh._handlers)
-        self.assertNotIn('scheme', dh._notconfigured)
+        self.assertIn('scheme', dh._notconfigured)
```

Comment:
There is one modification to expected behaviour here.
I believe it should be okay in this case.
Otherwise I can't negative-cache non-existing handlers in `notconfigured`, only existent but failing ones.